File: python/57InsertInterval.py
class Solution:
    def insert(self, intervals: List[List[int]], newInterval: List[int]) -> List[List[int]]:
        s,e = newInterval
        l,r=[],[]
        for i in intervals:
            if i[1] < s:
                l.append(i)
            elif i[0]> e:
                r.append(i)
            else:
                s=min(s,i[0])
                e=max(e,i[1])
        l.append([s,e])
        for j in r:
            l.append(j)
        return l

File: rust/57InsertInterval.rs
impl Solution {
	pub fn insert(intervals: Vec<Vec<i32>>, new_interval: Vec<i32>) -> Vec<Vec<i32>> {
			use std::cmp::{min, max};
			let mut ans = vec![];
			let mut right = vec![];
			let n = intervals.len();
			let mut s = new_interval[0];
			let mut e = new_interval[1];
			for curr in intervals {
					if curr[1] < new_interval[0] {
							ans.push(curr)
					} else if curr[0] > new_interval[1] {
							right.push(curr);
					} else {
							s = min(s, curr[0]);
							e = max(e, curr[1]);
					}
			}
			ans.push(vec![s,e]);
			ans.append(&mut right);
			ans
	}
}